Required elements for publishing

awead opened the pull request:
License and deposit agreement are now marked required, which will inform users using HTML5 feature and will not longer display the error messages if they are navigating to the page for the first time.

We have also updated the visibility label to read _Access_, making it consistent throughout the application.

Fixes #700 
Fixes #707

mtribone commented:
@cam156 👍

mtribone commented on spec/features/generic_work/upload_and_delete_spec.rb:
```diff
@@ -58,9 +58,13 @@
         click_link("Additional fields")
         expect(page).to have_no_css("#generic_work_contributor")
 
+        # Check for optional metadata
```
@cam156 I've basically just added a click, but did not add any tests for content on each tab. Yes?

awead committed:
Required elements for publishing

License and deposit agreement are now marked required, which will
inform users using HTML5 feature and will not longer display the error
messages if they are navigating to the page for the first time.

We have also updated the visibility label to read _Access_, making it
consistent throughout the application.

awead commented:
@rschenk I can't remember if there was a specific reason we were doing a `publish && validate` at the outset. Maybe it was to show the errors right away? But now we don't want to do that...

rschenk commented:
@awead @srerickson the point of the `publish && validate` at the beginning was to preemptively show any publication-level validation errors, _but at the time that was written, there was nothing on this form that the user had not already seen before (that had validations)._ I agree that #700 is a major UX issue, but I also don't think we should necessarily throw the baby out with the bath water. 

For example, `description` and `publication_date` are not required to save a draft but are required to publish, so you would see this when the form was first loaded.

![Untitled](https://user-images.githubusercontent.com/14730/99269264-2919eb00-27f4-11eb-9db1-c6525f7a6b54.png)

Another way to accomplish this would be below. It's a little more brittle but I think the UX is better:

```ruby
      def edit
        @work_version = WorkVersion
          .includes(file_version_memberships: [:file_resource])
          .find(params[:work_version_id])
        authorize(@work_version)

        @work_version.publish
        @work_version.validate

        # Remove any errors messages from fields that the user hasn't seen before
        # such as the "Publishing Details" section
        @work_version.errors.delete(:rights)
      end
```

awead commented:
@rschenk Wouldn't the above solution remove an actual error? If they hit publish, but hadn't selected rights, it would redirect back to edit and remove the error message?

rschenk commented:
@awead Not quite. If they hit publish but hadn't selected rights, it will _render_ the edit form, but it will not _redirect_ to the edit action, therefore none of the code in the `#edit` method will be executed even though the view is rendered. Kind of confusing.

rschenk reviewed (approved):
😎
